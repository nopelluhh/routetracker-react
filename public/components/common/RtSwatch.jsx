import React, {Component} from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import color from 'services/colorService'

class RtSwatch extends Component {
	render() {
		let c = this.props.color
		if (!this.props.color.color) {
			let t = this.props.colors.filter(color => color.value === this.props.color.toLowerCase())[0]
			c = {
				color: t? t.color : '#FFF',
				value: this.props.color.toLowerCase()
			}
		}

		const localStyle = {
			backgroundColor: c.color || 'white',
			color: color.textColor(c.color),
			borderRadius: '3px',
			height: '30px'
		}

		const style = Object.assign({}, localStyle, this.props.style? this.props.style : {})

		return (<span className={this.props.className} style={ style }><span className="hidden-sm-down">{ c.value || '' }</span></span>)
	}

}

RtSwatch.propTypes = {
	color: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
	colors: PropTypes.array,
	style: PropTypes.object
}

function mapState(state) {
	return {
		colors: state.team.colors
	}
}

export default connect(mapState)(RtSwatch)

export { RtSwatch }