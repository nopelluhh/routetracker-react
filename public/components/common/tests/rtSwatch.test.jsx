import { RtSwatch } from '../RtSwatch'

import React from 'react'
import renderer from 'react-test-renderer'

describe('rtSwatch tests', () => {
	it('should adjust style based on props', () => {
		const component = renderer.create(
            <RtSwatch color={'black'} colors={[{value: 'white', color: '#FFF'}, {value: 'black', color: '#000'}]}> Facebook </RtSwatch>
        )

		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
