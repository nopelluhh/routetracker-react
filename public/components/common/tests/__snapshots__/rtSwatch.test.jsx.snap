// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rtSwatch tests should adjust style based on props 1`] = `
<span
  className={undefined}
  style={
    Object {
      "backgroundColor": "#000",
      "borderRadius": "3px",
      "color": "#fff",
      "height": "30px",
    }
  }
>
  black
</span>
`;
